examples/common.h

```diff
@@ -0,0 +1,211 @@
+        return false;
+    }
+    return true;
+}
+
+po::options_description get_common_asymmetric_options_description()
+{
+    po::options_description commonAsymmetricOptions("Common options for asymmetric cryptography");
+    commonAsymmetricOptions.add_options()
+            ("private-key", po::value<std::string>(), "Path to the private key for decryption/signing")
+            ("private-key-password", po::value<std::string>()->default_value(""), "Optional: password for the private key")
+            ("public-key", po::value<std::string>(), "Path to the public key for encryption/verifying")
+            ("cert", po::value<std::string>(), "Path to the certificate used for encryption/verifying")
+            ;
+    return commonAsymmetricOptions;
+}
+
+bool isGetCommonAsymmetricOptionsValid(const po::variables_map &vm)
+{
+    if (vm.count("decrypt") || vm.count("sign")) {
+        if (!vm.count("private-key")) {
+            return false;
+        }
+    } else {
+        if (!(vm.count("public-key") xor vm.count("cert"))) {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool parseCommandLineOptions(const po::options_description &cmdlineOptions, po::variables_map &vm,
+                             int argc, char *argv[])
+{
+    try {
+        po::store(po::parse_command_line(argc, argv, cmdlineOptions), vm);
+    } catch (boost::wrapexcept<boost::program_options::invalid_command_line_syntax> &e) {
+        std::cerr << "Failure parsing the input. Please check your input!" << std::endl;
+        std::cerr << e.what();
+        return false;
+    } catch (boost::wrapexcept<boost::program_options::invalid_option_value> &e) {
+        std::cerr << "Got invalid option value (e.g. string for integer). Please check your input!" << std::endl;
+        return false;
+    } catch (boost::wrapexcept<boost::program_options::unknown_option> &e) {
+        std::cerr << "Unknown option selected: ";
+        std::cerr << e.what() << std::endl;
+        return false;
+    } catch (boost::exception &e) {
+        std::string info = boost::diagnostic_information(e);
+        std::cerr << "Caught boost exception during parsing. Please check your input!" << std::endl;
+        std::cerr << info << std::endl;
+        return false;
+    }
+    po::notify(vm);
+    return true;
+}
+
+void printAllParsedOptions(const po::variables_map &vm)
+{
+    for (const auto& it : vm) {
+        std::cout << it.first.c_str() << " ";
+        auto& value = it.second.value();
+        if (auto v = boost::any_cast<uint32_t>(&value))
+            std::cout << *v;
+        else if (auto v = boost::any_cast<std::string>(&value))
+            std::cout << *v;
+        else if (auto v = boost::any_cast<bool>(&value)) {
+            if (*v == true)
+                std::cout << "True";
+            else if (*v == false)
+                std::cout << "False";
+            else
+                std::cout << "error on boolean";
+        } else
```

Comment:
Please add brackets everywhere. The branches without brackets are a bug waiting to happen.